client/state/themes/selectors.js

```diff
@@ -695,3 +696,14 @@ export function shouldFilterWpcomThemes( state, siteId ) {
 		! isJetpackSiteMultiSite( state, siteId )
 	);
 }
+
+/**
+ * Returns the URL for purchasing a plan for the current site.
+ *
+ * @param  {Object}  state   Global state tree
+ * @param  {Number}  siteId  Site ID for which to buy the theme
+ * @return {?String}         Plan purchase URL
+ */
+export function getPlanUpgradeUrl( state, siteId ) {
+	return '/plans/' + getSiteSlug( state, siteId );
```

Comment:
What about changing the implementation to

```js
	if ( ! isJetpackSite( state, siteId ) || ! isThemePremium( state, themeId ) ) {
		return null;
	}
	return `/checkout/${ getSiteSlug( state, siteId ) }/professional`;
```

? That would send the user directly to checkout, with the JP Pro plan in their cart. That way, it'd be a bit more similar to how individual theme purchase is handled.